mpl_interactions/generic.py

```diff
@@ -306,18 +306,71 @@ class panhandler:
     """
     Enable panning a plot with any mouse button.
 
-    button determines which button will be used (default right click)
-    Left: 1
-    Middle: 2
-    Right: 3
+    .. code-block:: python
+
+       handler = panhandler(my_figure)
+
+       # Let it be disabled and garbage collected
+       handler = None
+
+    Parameters
+    ----------
+    button : int
+        Determines which button will be used (default right click).
+        Left: 1
+        Middle: 2
+        Right: 3
     """
 
     def __init__(self, fig, button=3):
         self.fig = fig
         self._id_drag = None
         self.button = button
-        self.fig.canvas.mpl_connect("button_press_event", self.press)
-        self.fig.canvas.mpl_connect("button_release_event", self.release)
+        self._id_press = None
+        self._id_release = None
+
+        self.enable()
+
+    @property
+    def enabled(self) -> bool:
+        """
+        Status of the panhandler, whether it's enabled or disabled.
+        """
+        return self._id_press is not None and self._id_release is not None
+
+    def enable(self):
+        """
+        Enable the panhandler. It should not be necessary to call this function
+        unless it's used after a call to :meth:`panhandler.disable`.
+
+        Raises
+        ------
+        RuntimeError
+            If the panhandler is already enabled.
+        """
+        if self.enabled:
+            raise RuntimeError("The panhandler is already enabled")
```

Comment:
How do you feel about just being a no-op in the case of already being enabled. 

Im open to an argument for why raising an error is better, but my gut says if we're being asked to enable and already are then it's fine to just go along with it.

Reply:
Hmm I don't have a strong opinion about it, but I'm mildly in favour of raising an exception because if the user is trying to enable/disable this twice, that's probably unintentional and a bug in their code.

Reply:
(that actually was the case for me once)

Reply:
fair enouguh. if it's an issue can always change it later.